doc/reference/trd-appid.md

```diff
@@ -0,0 +1,651 @@
+(described below). Kernels using Short IDs to test collisions between
+application identifiers SHOULD implement `Compress` in a manner that
+minimizes cases when two different valid application identifiers
+compress to the same Short ID (e.g., taking the low-order bits
+of a strong cryptographic hash function).
+
+In cases when a process does not have any application
+credentials, the verifier policy MAY assign it a global or local
+application identifier. If the verifier policy does not assign a
+process binary an application identifier then the kernel MUST NOT load
+or run that process.
+
+Consider these five use cases.
+
+  1. A process binary that has no application credentials: it only
+  runs on kernels that are willing to load applications without
+  credentials (e.g., research systems). The verifier policy defines
+  that process binaries with no credentials have a local application 
+  identifier of a SHA256 hash of the application binary. 
+
+  1. The verifier policy defines that the global application
+  identifier of a process is the public key used to sign its process
+  binary. Application credentials for a process binary consist of a
+  TBF header containing this public key as well as a signature for the
+  process binary signed with the associated private key. Before
+  verifiying a signature in a TBF header, the verifier policy decides
+  whether to it accepts the associated public key. The verifier policy
+  assigns a global application identifier as the public key in the TBF
+  header.
+
+  1. Multiple separate process binaries that run concurrently need to
+  be signed with a single public key. Each process binary is
+  identified by a unique identifier I. Application credentials for
+  these process binaries consist of a TBF header containing an ECDSA
+  public key, the unique identifier I, and a signature over the
+  process binary and I signed with the private key corresponding to
+  the public key in the header. The kernel decides whether to accept a
+  particular public key for verification. The verifier policy assigns
+  a global application identifier as the concatenation of the public
+  key and I.
+
+  1. A Tock system wants to load the same process binary in 
+  two different processes at the same time. It cannot. Every process
+  binary has a single application identifier, and Tock will not run
+  two processes with the same application identifier.
+
+  1. A Tock system wants to load the same application binary
+  in two different processes at the same time. The system administrator
+  installs two process binaries on the device, which contain the
+  same application binary. The process binaries have no credentials.
+  The verifier policy assigns a local application identifier to each
+  process binary based on its order in application flash.
+
+An application identifier provides an identity for an application
+binary. It allows the Tock kernel to know about the provenance and
```

Comment:
In the terminology section you say: `Every process binary has a single application identifier`. Is the application identifier from the application binary or the process binary?

Reply:
The application identifier is from the process binary.

I think the terminology is confusing. I am going to rethink it.